mesonbuild/modules/python.py

```diff
@@ -162,31 +162,19 @@ def extension_module_method(self, args: T.Tuple[str, T.List[BuildTargetSource]],
         new_deps = mesonlib.extract_as_list(kwargs, 'dependencies')
         pydep = next((dep for dep in new_deps if isinstance(dep, _PythonDependencyBase)), None)
         if pydep is None:
-            pydep = self._dependency_method_impl({})
+            pydep = self._dependency_method_impl({'limited_api': kwargs.pop('limited_api', '')})
             if not pydep.found():
                 raise mesonlib.MesonException('Python dependency not found')
             new_deps.append(pydep)
             FeatureNew.single_use('python_installation.extension_module with implicit dependency on python',
                                   '0.63.0', self.subproject, 'use python_installation.dependency()',
                                   self.current_node)
 
-        limited_api_version = kwargs.pop('limited_api')
         allow_limited_api = self.interpreter.environment.coredata.get_option(OptionKey('python.allow_limited_api'))
-        if limited_api_version != '' and allow_limited_api:
+        if pydep.limited_api != '' and allow_limited_api:
```

Comment:
There is quite a bit of follow-up logic here for Windows usage that I would really expect to be handled by .dependency() too, surely?

Reply:
I think I agree? The main reason I didn't move it already is that there's this bit:

```python
pydep_copy = copy.copy(pydep)
pydep_copy.find_libpy_windows(self.env, limited_api=True)
...
new_deps.remove(pydep)
new_deps.append(pydep_copy)
```

I think this is ensuring that the dependency has a Windows limited API libpy, even if the user provided one without. The modification is done on a copy to avoid modifying user inputs.

If we upstreamed the `find_libpy_windows` to the dependency, then a user could pass a non-limited API dependency that wouldn't get fixed. That's already the case on Linux, but it would be a "regression" on Windows. It seems niche/weird enough to me that I think we could just declare that unsupported, though.